doc/source/ray-core/patterns/map-reduce.rst

```diff
@@ -0,0 +1,37 @@
+Pattern: Using tasks to execute MapReduce
+=========================================
+
+In this pattern, Ray remote tasks can be used to implement the MapReduce paradigm.
+For the ``map`` stage, we can submit multiple tasks to process data in a distributed and parallel fashion.
+For the ``reduce`` stage, we use task arguments to fetch the results of each of these map tasks implicitly and aggregate on them.
+
+.. note::
+
+  It's generally recommended to use the high level Ray :ref:`Datasets <datasets>` library for data transformations rather than building MapReduce directly with Ray tasks.
+  For example, you can use Datasets APIs like :meth:`map_batches <ray.data.Dataset.map_batches>`, :meth:`aggregate <ray.data.Dataset.aggregate>` and :meth:`groupby <ray.data.Dataset.groupby>`.
+
+Example use case
+----------------
+
+Use Ray tasks to calculate the sum of a doubled list of integers in a MapReduce fasion.
+
+.. figure:: ../images/map-reduce.svg
+
+    Map and reduce
```

Comment:
Can we provide a bit more explanation of what is happening in the figure? Also, could we make the figure a single-round reduce instead of tree? Tree reduce is not as common, I think, and you can mention how it would look in the caption instead of showing it.